### ISSUE TITLE
[BUG]Failed to automatically obtain an IP address, resulting in inaccessibility after deployment.

### ISSUE THREAD

**Issue by maojunxyz** [metersphere/metersphere#20424]
### MeterSphere 版本

2.4.0

### 使用外置数据库

否

### 问题描述

The HOSTNAME variable is not set. Defaulting to a blank string.

### 重现步骤

root@20221129-instance:/opt/metersphere-offline-installer-v2.4.0# ./install.sh 
安装目录为 /opt/metersphere, 开始进行安装
[MeterSphere Log]: 拷贝安装文件到目标目录 
'./metersphere/docker-compose-api-test.yml' -> '/opt/metersphere/docker-compose-api-test.yml'
'./metersphere/docker-compose-project-management.yml' -> '/opt/metersphere/docker-compose-project-management.yml'
'./metersphere/docker-compose-data-streaming.yml' -> '/opt/metersphere/docker-compose-data-streaming.yml'
'./metersphere/docker-compose-kafka.yml' -> '/opt/metersphere/docker-compose-kafka.yml'
'./metersphere/docker-compose-workstation.yml' -> '/opt/metersphere/docker-compose-workstation.yml'
'./metersphere/docker-compose-redis.yml' -> '/opt/metersphere/docker-compose-redis.yml'
'./metersphere/docker-compose-minio.yml' -> '/opt/metersphere/docker-compose-minio.yml'
'./metersphere/docker-compose-prometheus.yml' -> '/opt/metersphere/docker-compose-prometheus.yml'
'./metersphere/docker-compose-node-controller.yml' -> '/opt/metersphere/docker-compose-node-controller.yml'
'./metersphere/docker-compose-mysql.yml' -> '/opt/metersphere/docker-compose-mysql.yml'
'./metersphere/docker-compose-eureka.yml' -> '/opt/metersphere/docker-compose-eureka.yml'
'./metersphere/docker-compose-gateway.yml' -> '/opt/metersphere/docker-compose-gateway.yml'
'./metersphere/conf' -> '/opt/metersphere/conf'
'./metersphere/conf/my.cnf' -> '/opt/metersphere/conf/my.cnf'
'./metersphere/conf/version' -> '/opt/metersphere/conf/version'
'./metersphere/conf/redis.conf' -> '/opt/metersphere/conf/redis.conf'
'./metersphere/conf/prometheus' -> '/opt/metersphere/conf/prometheus'
'./metersphere/conf/prometheus/prometheus.yml' -> '/opt/metersphere/conf/prometheus/prometheus.yml'
'./metersphere/conf/metersphere.properties' -> '/opt/metersphere/conf/metersphere.properties'
'./metersphere/version' -> '/opt/metersphere/version' (backup: '/opt/metersphere/version.20221202-2029')
'./metersphere/docker-compose-seleniarm.yml' -> '/opt/metersphere/docker-compose-seleniarm.yml'
'./metersphere/docker-compose-base.yml' -> '/opt/metersphere/docker-compose-base.yml'
'./metersphere/docker-compose-test-track.yml' -> '/opt/metersphere/docker-compose-test-track.yml'
'./metersphere/docker-compose-system-setting.yml' -> '/opt/metersphere/docker-compose-system-setting.yml'
'./metersphere/docker-compose-performance-test.yml' -> '/opt/metersphere/docker-compose-performance-test.yml'
'./metersphere/docker-compose-report-stat.yml' -> '/opt/metersphere/docker-compose-report-stat.yml'
'./metersphere/logs' -> '/opt/metersphere/logs'
'./metersphere/logs/README.md' -> '/opt/metersphere/logs/README.md'
'./metersphere/docker-compose-ui-test.yml' -> '/opt/metersphere/docker-compose-ui-test.yml'
[MeterSphere Log]: ======================= 开始安装 ======================= 
[MeterSphere Log]: ... 离线安装 docker 
[MeterSphere Log]: ... 启动 docker 
[MeterSphere Log]: 检测到 Docker Compose 已安装，跳过安装步骤 
[MeterSphere Log]: 加载镜像 
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/project-management:v2.4.0
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/report-stat:v2.4.0
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/test-track:v2.4.0
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/gateway:v2.4.0
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/redis:6.2.6
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/node-exporter:latest
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/seleniarm-grid-all:4.1.4-20220519
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/ui-test:v2.4.0
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/data-streaming:v2.4.0
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/jmeter-master:5.5-ms3-jdk11
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/mysql:8.0.30
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/minio:latest
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/prometheus:latest
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/performance-test:v2.4.0
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/eureka:v2.4.0
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/api-test:v2.4.0
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/system-setting:v2.4.0
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/workstation:v2.4.0
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/node-controller:v2.4.0
Loaded image: registry.cn-qingdao.aliyuncs.com/metersphere/kafka:3.3.1
[MeterSphere Log]: 启动服务 

The HOSTNAME variable is not set. Defaulting to a blank string.
Removing network metersphere_ms-network
Network metersphere_ms-network not found.
Removing volume metersphere_ms-conf
Volume metersphere_ms-conf not found.
Removing volume metersphere_ms-logs
Volume metersphere_ms-logs not found.
Removing volume metersphere_ms-data
Volume metersphere_ms-data not found.
Removing volume metersphere_ms-mysql-data
Volume metersphere_ms-mysql-data not found.
Removing volume metersphere_ms-prometheus-data
Volume metersphere_ms-prometheus-data not found.
Removing volume metersphere_ms-prometheus-conf
Volume metersphere_ms-prometheus-conf not found.
Removing volume metersphere_ms-redis-data
Volume metersphere_ms-redis-data not found.
Removing volume metersphere_ms-minio-data
Volume metersphere_ms-minio-data not found.

The HOSTNAME variable is not set. Defaulting to a blank string.
Creating network "metersphere_ms-network" with driver "bridge"
Creating volume "metersphere_ms-conf" with default driver
Creating volume "metersphere_ms-logs" with default driver
Creating volume "metersphere_ms-data" with default driver
Creating volume "metersphere_ms-mysql-data" with default driver
Creating volume "metersphere_ms-prometheus-data" with default driver
Creating volume "metersphere_ms-prometheus-conf" with default driver
Creating volume "metersphere_ms-redis-data" with default driver
Creating volume "metersphere_ms-minio-data" with default driver
Creating eureka             ... done
Creating ms-node-controller ... done
Creating ms-prometheus      ... done
Creating minio              ... done
Creating redis              ... done
Creating gateway            ... done
Creating kafka              ... done
Creating mysql              ... done
Creating nodeexporter       ... done
Creating system-setting     ... done
Creating ms-data-streaming  ... done
Creating report-stat        ... done
Creating project-management ... done
Creating performance-test   ... done
Creating api-test           ... done
Creating test-track         ... done

The HOSTNAME variable is not set. Defaulting to a blank string.
       Name                     Command                       State                                               Ports                                    
-----------------------------------------------------------------------------------------------------------------------------------------------------------api-test             /deployments/run-java.sh         Up (health: starting)   0.0.0.0:10000->10000/tcp, 0.0.0.0:10001->10001/tcp, 0.0.0.0:10002->10002/tcp,                                                                              0.0.0.0:10003->10003/tcp, 0.0.0.0:10004->10004/tcp, 0.0.0.0:10005->10005/tcp,                                                                              0.0.0.0:10006->10006/tcp, 0.0.0.0:10007->10007/tcp, 0.0.0.0:10008->10008/tcp,                                                                              0.0.0.0:10009->10009/tcp, 0.0.0.0:10010->10010/tcp                           
eureka               /deployments/run-java.sh         Up (healthy)                                                                                         
gateway              /deployments/run-java.sh         Up (healthy)            0.0.0.0:8081->8000/tcp                                                       
kafka                /opt/bitnami/scripts/kafka ...   Up (healthy)            0.0.0.0:9092->9092/tcp                                                       
minio                /usr/bin/docker-entrypoint ...   Up (healthy)            0.0.0.0:9000->9000/tcp, 0.0.0.0:9001->9001/tcp                               
ms-data-streaming    /deployments/run-java.sh         Up (healthy)                                                                                         
ms-node-controller   /deployments/run-java.sh         Up (healthy)            0.0.0.0:8082->8082/tcp, 0.0.0.0:9100->9100/tcp                               
ms-prometheus        /bin/prometheus --config.f ...   Up (healthy)            0.0.0.0:9090->9090/tcp                                                       
mysql                docker-entrypoint.sh mysqld      Up (healthy)            0.0.0.0:3307->3306/tcp, 33060/tcp                                            
nodeexporter         /bin/node_exporter --path. ...   Up (healthy)                                                                                         
performance-test     /deployments/run-java.sh         Up (health: starting)                                                                                
project-management   /deployments/run-java.sh         Up (health: starting)                                                                                
redis                docker-entrypoint.sh redis ...   Up (healthy)            0.0.0.0:6379->6379/tcp                                                       
report-stat          /deployments/run-java.sh         Up (health: starting)                                                                                
system-setting       /deployments/run-java.sh         Up (healthy)                                                                                         
test-track           /deployments/run-java.sh         Up (health: starting)                                                                                
======================= 安装完成 =======================

请通过以下方式访问:
 URL: http://$LOCAL_IP:8081
 用户名: admin
 初始密码: metersphere
企业用户升级后需手动设置 install.conf 中的 MS_UI_ENABLED=true 和 MS_WORKSTATION_ENABLED=true 并执行 'msctl reload' 来开启UI和工作台功能
您可以使用命令 'msctl status' 检查服务运行情况.

**Comment by liuruibin** [comment]
很抱歉长时间未处理，不知道问题是否依然存在，这里先把issue关闭，如果问题仍然存在请重新打开
